pts-core/objects/client/pts_logger.php

```diff
@@ -75,8 +75,17 @@ public function log($message, $date_prefix = true)
 		if($this->log_file == null)
 			return;
 
+		$traces = debug_backtrace();
+
+		if (isset($traces[0]))
+    		{
+		        $caller = $traces[1]['function'];
+		        $line = $traces[0]['line'];
+		        $file = basename($traces[0]['file']);
+		}
+
 		$message = pts_user_io::strip_ansi_escape_sequences($message);
-		file_put_contents($this->log_file, ($date_prefix ? '[' . date('M ' . str_pad(date('j'), 2, ' ', STR_PAD_LEFT) . ' H:i:s Y') . '] ' : null) . $message . PHP_EOL, FILE_APPEND);
+		file_put_contents($this->log_file, ($date_prefix ? '[' . date(DateTimeInterface::ISO8601) . '] ' : null) . "[" . $caller . "(". $file . ":" . $line . ")] " . $message . PHP_EOL, FILE_APPEND);
```

Comment:
This pull request looks fine except DateTimeInterface::ISO8601 should be changed out for 'Y-m-d\TH:i:sO' or similar. Per https://www.php.net/manual/en/class.datetimeinterface.php, DateTimeInterface was only added to PHP 5.5+ thus would break the existing PTS PHP 5.3~5.4 compatibility on some (aging) enterprise Linux distributions.